app/scaffold-controllers/navigationController.js

```diff
@@ -77,23 +84,21 @@ function(
                 searchBarController,
                 drawerIconEnabled
             );
-
-            Astro.events.on('welcome:shown', function() {
+            var handleActiveState = function() {
                 if (navigationController.isActive) {
                     Astro.events.once('welcome:hidden', function() {
                         navigationController.isActive = true;
                     });
                 }
                 navigationController.isActive = false;
+            };
+
+            Astro.events.on('welcome:shown', function() {
+                handleActiveState();
```

Comment:
What do you think about removing the wrapping function and passing in `handleActiveState` as the callback?